nixos/modules/services/networking/unifi.nix

```diff
@@ -110,27 +118,77 @@ in
       '';
     };
 
+    httpPort = mkOption {
+      type = types.port;
+      default = 8080;
+      description = ''
+        Port for AP inform
+      '';
+    };
+
+    httpsPort = mkOption {
+      type = types.port;
+      default = 8443;
+      description = ''
+        Port for HTTPS traffic to controller UI.
+      '';
+    };
+
+    stunPort = mkOption {
+      type = types.port;
+      default = 3478;
+      description = ''
+        Port for STUN traffic.
+      '';
+    };
+
+    enableHsts = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Enable HSTS.
+      '';
+    };
+
+    database.journaling = mkOption {
+      type = types.bool;
+      default = true;
+      description = ''
+        Use database journaling. Disable this to use less disk space in exchange for a higher risk of file corruption.
+      '';
+    };
+
+    systemProperties = mkOption {
+      type = types.attrs;
+      default = {};
+      description = ''
+        Key-value pairs that will be written to system.properties
+      '';
+    };
   };
 
   config = mkIf cfg.enable {
 
     users.users.unifi = {
-      uid = config.ids.uids.unifi;
       description = "UniFi controller daemon user";
-      home = "${stateDir}";
+      home = stateDir;
+      group = "unifi";
     };
 
+    users.groups.unifi = {};
+
     networking.firewall = mkIf cfg.openPorts {
       # https://help.ubnt.com/hc/en-us/articles/218506997
       allowedTCPPorts = [
-        8080  # Port for UAP to inform controller.
-        8880  # Port for HTTP portal redirect, if guest portal is enabled.
-        8843  # Port for HTTPS portal redirect, ditto.
-        6789  # Port for UniFi mobile speed test.
+        cfg.httpPort  # Port for UAP to inform controller.
+        cfg.httpsPort # Port for controller UI/API.
+        8880          # Port for HTTP portal redirect, if guest portal is enabled.
+        8843          # Port for HTTPS portal redirect, ditto.
```

Comment:
We should introduce config options for `portal.http.port` and `portal.https.port` as well.